File: manual/data/cpp/tests/test_tqdm_trange.cpp
#include <gtest/gtest.h>
#include "../src/function_tqdm_trange.cpp"

class TestRangeFunction : public ::testing::Test {
protected:
    void SetUp() override {}
    void TearDown() override {}
};

TEST_F(TestRangeFunction, TestTypeOfResult) {
    std::string result = ProgressRange::might_fail_function(0, 5, 1);
    EXPECT_EQ(typeid(result).name(), typeid(std::string).name());
}

TEST_F(TestRangeFunction, TestCustomMiniters) {
    std::string result = ProgressRange::might_fail_function(0, 5, 1, "Test", true, 2);
    EXPECT_EQ(result.length(), 5);
}

TEST_F(TestRangeFunction, TestCustomMininterval) {
    std::string result = ProgressRange::might_fail_function(0, 5, 1, "Test", true, 0, 0.0f, 0.1f);
    EXPECT_EQ(result.length(), 5);
}

TEST_F(TestRangeFunction, TestDescUpdated) {
    std::string result = ProgressRange::might_fail_function(0, 5, 1, "New description");
    EXPECT_EQ(result.length(), 5);
}

TEST_F(TestRangeFunction, TestAsciiFalse) {
    std::string result = ProgressRange::might_fail_function(0, 5, 1, "Test", false);
    EXPECT_EQ(result.length(), 5);
}

File: manual/data/java/tests/test_tqdm_trange.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_tqdm_trange {

    @Test
    void test_type_of_result() {
        function_tqdm_trange instance = new function_tqdm_trange();
        String result = instance.might_fail_function(0, 5, 1, "General trange loop", true, null, null, null);
        assertEquals(String.class, result.getClass());
    }

    @Test
    void test_custom_miniters() {
        function_tqdm_trange instance = new function_tqdm_trange();
        String result = instance.might_fail_function(0, 5, 1, "General trange loop", true, 2, null, null);
        assertEquals(5, result.length());
    }

    @Test
    void test_custom_mininterval() {
        function_tqdm_trange instance = new function_tqdm_trange();
        String result = instance.might_fail_function(0, 5, 1, "General trange loop", true, null, null, 0.1f);
        assertEquals(5, result.length());
    }

    @Test
    void test_desc_updated() {
        function_tqdm_trange instance = new function_tqdm_trange();
        String result = instance.might_fail_function(0, 5, 1, "New description", true, null, null, null);
        assertEquals(5, result.length());
    }

    @Test
    void test_ascii_false() {
        function_tqdm_trange instance = new function_tqdm_trange();
        String result = instance.might_fail_function(0, 5, 1, "General trange loop", false, null, null, null);
        assertEquals(5, result.length());
    }
}